src/HPolyhedron.jl

```diff
@@ -484,11 +484,13 @@ function remove_redundant_constraints!(P::PT;
         br = b[non_redundant_indices]
         br[i] = b[j] + one(N)
         lp = linprog(-α, Ar, '<', br, -Inf, Inf, backend)
-        if lp.status != :Optimal
-            error("LP is not optimal")
+        if lp.status == :Infeasible
+            throw(DomainError(P, "the polyhedron is empty"))
```

Comment:
perhaps this should instead throw an assertion error, to be checked from the outside (e.g. from `intersection`)

Reply:
What is "the polyhedron"? And why should an empty intersection not be allowed?
I think this case should just return an `EmptySet`.

Reply:
i threw an error because `remove_redundant_constraints!` is supposed to modify an `HPolyhedron` in place. changing `P` to an emptyset breaks type stability.

otherwise we could change the signature to something like:

```julia
function remove_redundant_constraints!(P::Union{PT<:HPoly{N}, EmptySet{N}},
                                       backend=GLPKSolverLP()) where {N}
....
```

but this could break type stability in other places, no? like functions that use `remove_redundant_constraints!` should be aware that it can return empty sets.

so just throwing an error didn't seems ok to me.

Reply:
Oh, I thought this was in `intersection`. But why would you have to throw an error? Can we not just print a warning and return the original set?